pkg/content/utils.go

```diff
@@ -1,9 +1,163 @@
 package content
 
-import ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	digest "github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+)
 
 // ResolveName resolves name from descriptor
 func ResolveName(desc ocispec.Descriptor) (string, bool) {
 	name, ok := desc.Annotations[ocispec.AnnotationTitle]
 	return name, ok
 }
+
+// tarDirectory walks the directory specified by path, and tar those files with a new
+// path prefix.
+func tarDirectory(root, prefix string, w io.Writer) error {
+	tw := tar.NewWriter(w)
+	defer tw.Close()
+	if err := filepath.Walk(root, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		// Rename path
+		name, err := filepath.Rel(root, path)
+		if err != nil {
+			return err
+		}
+		name = filepath.Join(prefix, name)
+		name = filepath.ToSlash(name)
+
+		// Generate header
+		var link string
+		mode := info.Mode()
+		if mode&os.ModeSymlink != 0 {
+			if link, err = os.Readlink(path); err != nil {
+				return err
+			}
+		}
+		header, err := tar.FileInfoHeader(info, link)
+		if err != nil {
+			return errors.Wrap(err, path)
+		}
+		header.Name = name
+		header.Uid = 0
```

Comment:
Could you explain the uid/gid/uname/gname here? Should we just rely on defaults for these?

Reply:
They are all information about local machine.
`uid` --> user id in `/etc/passwd`
`uname` --> user name in `/etc/passwd`
`gid` --> group id in `/etc/group`
`gname` --> group name in `/etc/group`

They are meaningful when using `tar` as a backup method, and they are meaningless and leaking privacy when for sharing